src/edu/csus/ecs/pc2/core/execute/SandboxUtilities.java

```diff
@@ -23,28 +23,35 @@
      * 
      * @param executeDirectory
      * @param run
+     * @param sourceFileName - name of team's source file name, ex. Sumit.java
      * @param problem
-     * @param inputFile
+     * @param language
+     * @param judgesDataFile - name of judge's data file
      * @param memlimitK
      * @param timelimitMS
-     * @return filename were properties written
+     * @return filename where properties written.
      * @throws IOException
      */
-    public static String writeSandboxInputProperties(String executeDirectory, Run run, Problem problem, Language language, File inputFile, int memlimitK, int timelimitMS) throws IOException {
+    public static String writeSandboxInputProperties(String executeDirectory, Run run, String sourceFileName, Problem problem, Language language, File judgesDataFile, int memlimitK, int timelimitMS) throws IOException {
 
         Properties properties = new Properties();
 
         properties.put(SandboxInput.RUNID_KEY, Integer.toString(run.getNumber()));
 
-        properties.put(SandboxInput.MEMLIMIT_KEY, inputFile);
-        properties.put(SandboxInput.TIMELIMIT_KEY, timelimitMS);
+        if (memlimitK > 0) {
+            properties.put(SandboxInput.MEMLIMIT_KEY, Integer.toString(memlimitK));
```

Comment:
Memory limits are almost always expressed these days in Mb, not Kb.  This variable name is either very misleading (if the actual data is really a value in Mb), or else the intent is bad -- that is, if the intent really is to force people to specify memory limits in units of Kb.  This would mean for example that if I want to specify a memory limit of 4GB I would have to write it as 4096000.  There is a reason the CLICS spec specifies memory limit values as being Mb, and I think it's a good one that we should stick to.

Reply:
changed to memlimitM